### DISCUSSION TITLE
Only allow Single Role for Users

### DISCUSSION THREAD

**Discussion by martingrondein-nls** [spatie/laravel-permission#2237] [Q&A]
Currently it seems a User can have multiple roles.  How do I restrict/change `laravel-permission` to only allow 1 role per user?

**Comment by erikn69** [comment]
Looks a lot like https://github.com/spatie/laravel-permission/discussions/2236

You can customize almost everything, this is agnostic, you have to add your own  logic for assign only one role per user

**Reply by ashek1412** [reply]
hi, is there any example

**Comment by drbyte** [comment]
> Currently it seems a User can have multiple roles. How do I restrict to only allow 1 role per user?

Some options:
(Note that the `$user->roles` property is an Eloquent Collection of roles that are assigned to the user)
- in your application, wherever you offer a button to assign more roles, you could check what roles that user already has, and if they have one already, don't offer the button
- in your application controller, wherever you assign the role after someone clicks a button to assign a role, you can first check what roles the user has, and throw an error if it is attempting to assign too many roles

**Comment by craigrileyuk** [comment]
Note that the `HasRoles` trait also comes with a `syncRoles` function.

If you use this rather than `assignRole`, it will automatically remove any other roles before assigning the one given.
